torch/csrc/cuda/nccl.cpp

```diff
@@ -70,7 +80,7 @@ static ArrayRef<ncclComm_t> _get_communicators(TensorList inputs, ArrayRef<ncclC
   }
 }
 
-static void _check_inputs(std::vector<at::Tensor> &inputs, std::vector<at::Tensor> &outputs, int input_multiplier, int output_multiplier) {
+static void _check_inputs(TensorList inputs, TensorList outputs, int input_multiplier, int output_multiplier) {
```

Comment:
They used to be references and now are passed by value. Is that expected?

Reply:
`TensorList` is intended to be passed by value (just a pointer and size). It is typedef of `ArrayRef<Tensor>`.

Reply:
Right, I always forget that ATen typedefs are `ArrayRef`s.

Reply:
But remember that `tensor_list` is a vector of tensors ;) (our naming convention is awful)